source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -869,6 +873,61 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
   }
 }
 
+void ServerContextImpl::populateServerNamesMap(TlsContext& ctx, int pkey_id) {
+  if (ctx.cert_chain_ == nullptr) {
+    return;
+  }
+
+  auto populate = [&](const std::string& sn) {
+    std::string sn_pattern = sn;
+    if (absl::StartsWith(sn, "*.")) {
+      sn_pattern = sn.substr(1);
+    }
+    PkeyTypesMap pkey_types_map;
+    auto sn_match = server_names_map_.try_emplace(sn_pattern, pkey_types_map).first;
```

Comment:
You're ignoring try_emplace().second, which is a bool indicating whether the insertion happened. If it didn't happen, this means there's a duplicate cert, and an error should be thrown.

Reply:
No, we do not need to try_emplace().second and check raise error at this point, We support multiple certs for one server name pattern.

For instance, if we have a ecdsa cert for "www.example.com", we can add one more rsa cert for it.

Reply:
Got it; that makes sense. Can you add a comment clarifying when it's expected for it to already exist?

Reply:
Yes, will update.